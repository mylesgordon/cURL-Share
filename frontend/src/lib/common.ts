export function splitAndTrim(input: string): Array<string> {
		return input.split(',').map((item) => item.trim());
	}