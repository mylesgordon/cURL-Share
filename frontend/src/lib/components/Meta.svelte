<script lang="ts">
	export let title = 'Unknown';
</script>

<svelte:head>
	<title>{title} | cURL Share</title>
</svelte:head>
