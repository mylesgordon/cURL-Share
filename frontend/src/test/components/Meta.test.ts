import Meta from '../../lib/components/Meta.svelte';
import { render } from '@testing-library/svelte';
import { testMatchingSnapshot } from '../common';

describe('Meta component', () => {
	testMatchingSnapshot(Meta);

	it('should default to Unknown without an argument', () => {
		render(Meta);
		expect(document.title).toBe("Unknown | cURL Share");
	} )

	it('should use the title argument', () => {
		render(Meta, { title: "Wow, what a page!"});
		expect(document.title).toBe("Wow, what a page! | cURL Share");
	} )
});
