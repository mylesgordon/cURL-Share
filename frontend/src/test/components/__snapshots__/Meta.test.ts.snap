// Vitest Snapshot v1

exports[`Meta component > should match snapshot 1`] = `
<body>
  <div>
    <!--&lt;Meta&gt;-->
  </div>
</body>
`;
